### PR TITLE
pgrep: Add -d option to specify a pid delimiter

### PR DESCRIPTION
**Pull request by tcl3**
This is useful for commands which expect a comma-separated list of pids.

Example usage (list processes starting with S):

pgrep_delimiter

### CODE REVIEW

File: Userland/Utilities/pgrep.cpp [hunk 1]
```diff
@@ -19,11 +19,13 @@ ErrorOr<int> serenity_main(Main::Arguments args)
     TRY(Core::System::unveil("/etc/passwd", "r"));
     TRY(Core::System::unveil(nullptr, nullptr));
 
+    StringView pid_delimiter { "\n"sv };
```
Comment (caoimhebyrne) [inline Userland/Utilities/pgrep.cpp:22]: Why are we using `{ ... }` here?
```cpp
StringView pid_delimiter = "\n"sv;
```

File: Userland/Utilities/pgrep.cpp [hunk 2]
```diff
@@ -50,9 +52,18 @@ ErrorOr<int> serenity_main(Main::Arguments args)
...
-        outln("{}", match);
+        if (displayed_at_least_one)
+            out("{}{}"sv, pid_delimiter, match);
+        else
+            out("{}"sv, match);
+
+        displayed_at_least_one = true;
     }
 
+    if (displayed_at_least_one)
```
Comment (caoimhebyrne) [inline Userland/Utilities/pgrep.cpp:65]: This feels a bit awkward... why do we do this?

**Review by caoimhebyrne — CHANGES_REQUESTED** [review]

File: Userland/Utilities/pgrep.cpp (hunk 2 above)
Comment (tcl3) [reply Userland/Utilities/pgrep.cpp:65]: This ensures that the output ends with a newline, even when we don't use the default newline delimiter. `pidof`  has the same logic.

Comment (caoimhebyrne) [reply Userland/Utilities/pgrep.cpp:65]: Sure, that makes sense :)

**Review by caoimhebyrne — APPROVED** [review]
Looks good to me :^)

**Pull request merged**